src/server.c

```diff
@@ -4891,7 +4891,7 @@ void version(void) {
 }
 
 void usage(void) {
-    fprintf(stderr,"Usage: ./redis-server [/path/to/redis.conf] [options]\n");
+    fprintf(stderr,"Usage: ./redis-server [/path/to/redis.conf] [options] [-]\n");
     fprintf(stderr,"       ./redis-server - (read config from stdin)\n");
```

Comment:
isn't that a documentation of allowing `-` as a first argument to load from stdin?
i guess you meant that it wasn't listed in the above "Usage" line, only in the examples.
but i'd say that seeing this example and the other examples, as a user i would have concluded i can pass additional options after it.

maybe we are allowed to break this argument syntax, but probably only in a major version.
i suppose we should avoid it, and it doesn't look very ugly to me (i.e. the first argument can either be a config file, a `--` option or a `-` indicating stdin instead of a config file)

Reply:
to me this text seems to suggest that the first argument can either be a config file or a `-`.
i don't think it means that if the first argument is `-` then the second argument can be a config file.

considering that, i think that it's enough to just check if the first argument is a `-` then there's no config file, and we set the flag for stdin to true and resume parsing normally.
for all i care the `-` can be also added as a last argument too (the stdin flag would already be true)

i'd rather do that than try to consider if / when we can break old syntax.

Reply:
Ok - So:
If '-' is first argument --> no config file - only stdin, options can be provided after
If someone wants both stdin & config file, then '-' has to be last

will update.

Reply:
@oranagra updated.